src/components/Authentication/Login.vue

```diff
@@ -0,0 +1,104 @@
+<template lang="pug">
+.login-widget
+  v-alert.mt-0(
+      dismissible,
+      transition="scale-transition",
+      v-for="err in alerts.errors",
+      :key="err",
+      :value="!!err",
+      type="error") {{ err }}
+  v-container
+    v-form(@submit.prevent="login", ref="login")
+      v-text-field(
+          v-model="username",
+          label="Username or e-mail",
+          autofocus,
+          :rules="nonEmptyRules",
+          prepend-icon="$vuetify.icons.user",
+          type="text")
+      v-text-field(
+          v-model="password",
+          type="password",
+          label="Password",
+          :rules="nonEmptyRules",
+          prepend-icon="$vuetify.icons.lock")
+      v-layout(row)
+        v-btn.ml-0(type="submit",
+            color="primary",
+            :disabled="inProgress",
+            :loading="inProgress")
+          v-icon.aligned(left) $vuetify.icons.login
+          | Login
+        v-spacer
+        v-btn(flat, color="primary", @click="forgotPasswordAction") Forgot Password?
```

Comment:
I'd suggest changing this to

```pug
v-btn(
    flat, color="primary", :to="forgotPasswordRoute", :href="forgotPasswordUrl",
    @click="$emit('forgotPassword')") Forgot Password?
```

And removing the `forgotPasswordAction()` method. The nice thing about this way would be that if a URL or route was passed, the button becomes a link and users can open it in a new tab, copy the URL with right click, etc.